service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/flight/RetrieveGcpControlledBigQueryDatasetLifetimeStep.java

```diff
@@ -0,0 +1,145 @@
+package bio.terra.workspace.service.resource.controlled.cloud.gcp.flight;
+
+import static bio.terra.workspace.common.utils.FlightUtils.validateRequiredEntries;
+import static bio.terra.workspace.service.crl.CrlService.getBigQueryDataset;
+import static bio.terra.workspace.service.resource.model.WsmResourceType.CONTROLLED_GCP_BIG_QUERY_DATASET;
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.CONTROLLED_BIG_QUERY_DATASETS_WITHOUT_LIFETIME;
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_PARTITION_LIFETIME_MAP;
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.CONTROLLED_BIG_QUERY_DATASET_RESOURCE_ID_TO_TABLE_LIFETIME_MAP;
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.CONTROLLED_RESOURCE_ID_TO_WORKSPACE_ID_MAP;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.resource.controlled.cloud.gcp.bqdataset.ControlledBigQueryDatasetResource;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.google.api.services.bigquery.model.Dataset;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import javax.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RetrieveGcpControlledBigQueryDatasetLifetimeStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(RetrieveGcpResourcesRegionStep.class);
+  private final CrlService crlService;
+
+  public RetrieveGcpControlledBigQueryDatasetLifetimeStep(CrlService crlService) {
+    this.crlService = crlService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    validateRequiredEntries(
+        context.getWorkingMap(), CONTROLLED_BIG_QUERY_DATASETS_WITHOUT_LIFETIME);
+    List<ControlledResource> controlledBigQueryDatasets =
```

Comment:
we can just deserialize to List<ControlledGcpBqDatasetResource> here.

Reply:
Can you clarify what you mean by that?

Reply:
i'm sorry. i was thinking one thing and writing another.

List<ControlledBqDatasetResource> datasets
    = context
            .getWorkingMap()
            .get(CONTROLLED_BIG_QUERY_DATASETS_WITHOUT_LIFETIME, new TypeReference<>() {});

so we don't need to do the castbyEnum